lib/shopping_cart.rb

```diff
@@ -0,0 +1,60 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+  attr_accessor :products_in_cart
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @products_in_cart = 0
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {
+      name: "King Soopers", 
+      capacity: 30
+    }
+  end
+
+  def total_number_of_products
+    @products_in_cart = 0
+    @products.each do |product|
+      @products_in_cart += product.quantity
+    end
+    @products_in_cart
+  end
+
+  def is_full?
+    @products_in_cart >= 30
+  end
+
+  def products_by_category(requested_item)
+    sorted_products = []
+    @products.find_all do |product|
```

Comment:
This is the right enumerable! Just slightly misapplied. The `#find_all` enumerable should return a boolean value in the code block within the `do` and `end`.  So instead of what you have above we should put

```
@products.find_all do |product|
  product.category == requested_item
end
```